pkg/controllers/v1alpha1/databackup/implement.go

```diff
@@ -332,15 +348,49 @@ func (r *DataBackupReconcilerImplement) reconcileExecutingDataBackup(ctx reconci
 // generateDataBackupValueFile builds a DataBackupValueFile by extracted specifications from the given DataBackup, and
 // marshals the DataBackup to a temporary yaml file where stores values that'll be used by fluid dataBackup helm chart
 func (r *DataBackupReconcilerImplement) generateDataBackupValueFile(ctx reconcileRequestContext, masterPod *v1.Pod) (valueFileName string, err error) {
+	log := ctx.Log.WithName("generateDataBackupValueFile")
 	databackup := ctx.DataBackup
 	nodeName, ip, rpcPort := utils.GetAddressOfMaster(masterPod)
 
-	imageName, imageTag := docker.GetWorkerImage(r.Client, databackup.Spec.Dataset, "alluxio", databackup.Namespace)
+	targetDataset, err := utils.GetDataset(r.Client, databackup.Spec.Dataset, databackup.Namespace)
+	if err != nil {
+		return "", err
+	}
+
+	index, boundedRuntime := utils.GetRuntimeByCategory(targetDataset.Status.Runtimes, common.AccelerateCategory)
+	if index == -1 {
+		return "", errors.New("bounded runtime with Accelerate Category is not found on the target dataset")
+	}
+
+	var imageName, imageTag, javaEnv, runtimeType, imageEnv, defaultImage string
+	switch boundedRuntime.Type {
+	case common.ALLUXIO_RUNTIME:
+		imageName, imageTag = docker.GetWorkerImage(r.Client, databackup.Spec.Dataset, "alluxio", databackup.Namespace)
+		javaEnv = "-Dalluxio.master.hostname=" + ip + " -Dalluxio.master.rpc.port=" + strconv.Itoa(int(rpcPort))
+		runtimeType = "alluxio"
+	case common.GOOSEFS_RUNTIME:
+		imageName, imageTag = docker.GetWorkerImage(r.Client, databackup.Spec.Dataset, "goosefs", databackup.Namespace)
+		javaEnv = "-Dgoosefs.master.hostname=" + ip + " -Dgoosefs.master.rpc.port=" + strconv.Itoa(int(rpcPort))
+		runtimeType = "goosefs"
+	default:
+		log.Error(fmt.Errorf("RuntimeNotSupported"), "The runtime is not supported yet", "runtime", boundedRuntime)
+		r.Recorder.Eventf(&ctx.DataBackup,
+			v1.EventTypeNormal,
+			common.RuntimeNotReady,
+			"Bounded accelerate runtime not supported")
+	}
 
 	if len(imageName) == 0 {
```

Comment:
Can we wrap `docker.GetImageTagFromEnv`  and `docker.GetImageRepoFromEnv` in `docker.GetWorkerImage` function ? 
```
imageName, imageTag := docker.GetWorkerImage(r.Client, databackup.Spec.Dataset, runtimeType, databackup.Namespace)
image := fmt.Sprintf("%s:%s", imageName, imageTag)
```
Do we have the case where the image name and the image tag come from different places ? 
Or can it be avoided ?

Reply:
I think we get image info from configmap and host env, will double check. But i think it is too strict.